File: android/app/src/main/java/org/coepi/android/localstorage/room/ContactDao.java
package org.coepi.android.localstorage.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import org.coepi.android.localstorage.GeoTime;
import org.coepi.android.localstorage.room.Contact;

import java.util.List;

@Dao
public interface ContactDao {
    @Query("SELECT * FROM contact")
    List<Contact> getAll();

    @Query("SELECT * FROM contact WHERE uid IN (:uuIds)")
    List<Contact> loadAllByIds(String uuIds);

    @Query("SELECT * FROM contact WHERE :first <= timestamp AND " +
            "timestamp <= :last LIMIT 1")
    List<Contact> findByRange(String first, String last);

    @Insert
    void insertAll(Contact contact );

    @Delete
    void delete( Contact contact );
}

File: android/app/src/main/java/org/coepi/android/localstorage/room/ContactDao.kt
package org.coepi.android.localstorage.room

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import org.coepi.android.localstorage.Contact

@Dao
interface ContactDao {
    @get:Query("SELECT * FROM contact")
    val all: List<Contact?>?

    @Query("SELECT * FROM contact WHERE uuidHash in (:ids)")
    fun loadAllById(ids: String?): List<Contact?>?

    @Query("SELECT * FROM contact WHERE :first <= timeStamp AND timeStamp <= :last LIMIT 1")
    fun findByRange(first: Long?, last: Long?): List<Contact>?

    @Insert
    fun insert(contact: Contact?)

    @Delete
    fun delete(contact: Contact?)
}